src/Catalyst.Core.Modules.Consensus/Deltas/DeltaBuilder.cs

```diff
@@ -177,11 +218,61 @@ private IList<TransactionBroadcast> GetValidTransactionsForDelta(IList<Transacti
             //lock time equals 0 or less than ledger cycle time
             //we assume all transactions are of type non-confidential for now
 
-            var validTransactionsForDelta =
-                allTransactions.Where(m => m.IsPublicTransaction && m.HasValidEntries()).ToList();
-            var rejectedTransactions = allTransactions.Except(validTransactionsForDelta);
+            List<TransactionBroadcast> validTransactionsForDelta = new List<TransactionBroadcast>();
+            List<TransactionBroadcast> rejectedTransactions = new List<TransactionBroadcast>();
+
+            int allTransactionsCount = allTransactions.Count;
+            for (int i = 0; i < allTransactionsCount; i++)
+            {
+                TransactionBroadcast currentItem = allTransactions[i];
+                if (!IsTransactionOfAcceptedType(currentItem) || !currentItem.HasValidEntries())
+                {
+                    rejectedTransactions.Add(currentItem);
+                    continue;
+                }
+
+                validTransactionsForDelta.Add(currentItem);
+            }
+            
+            validTransactionsForDelta.Sort(AveragePriceComparer.InstanceDesc);
+
+            ulong totalLimit = 0UL;
+            int allValidCount = validTransactionsForDelta.Count;
+            int rejectedCountBeforeLimitChecks = rejectedTransactions.Count;
+            for (int i = 0; i < allValidCount; i++)
+            {
+                TransactionBroadcast currentItem = validTransactionsForDelta[i];
+                ulong remainingLimit = DeltaGasLimit - totalLimit;
+                if (remainingLimit < MinTransactionEntryGasLimit)
+                {
+                    for (int j = i; j < allValidCount; j++)
+                    {
+                        currentItem = validTransactionsForDelta[j];
+                        rejectedTransactions.Add(currentItem);    
+                    }
+                    
+                    break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4350015)